capture_service/device_mgr.cc

```diff
@@ -509,201 +640,169 @@ absl::Status DeviceManager::DeployReplayApk(const std::string &serial)
+                                                        kProfilingPluginName);
+    RETURN_IF_ERROR(m_device->Adb().Run(absl::StrCat("shell chmod +x ", binary_path_on_device)));
+
+    LOGD("RunReplayProfilingBinary(): RUN\n");
+    std::string metrics_str = absl::StrJoin(settings.metrics, " ");
+    std::string gfxr_replay_flag = settings.replay_flags_str.empty() ?
+                                   "" :
+                                   absl::StrFormat("--gfxr_replay_flags \"%s\"",
+                                                   settings.replay_flags_str);
+    std::string cmd = absl::StrFormat("shell %s %s %s %s",
+                                      binary_path_on_device,
+                                      settings.remote_capture_path,
+                                      gfxr_replay_flag,
+                                      metrics_str);
+    RETURN_IF_ERROR(m_device->Adb().Run(cmd));
+
+    LOGD("RunReplayProfilingBinary(): RETRIEVE ARTIFACTS\n");
+    std::filesystem::path parse_remote_path = settings.remote_capture_path;
+    std::string           csv_local_name = absl::StrFormat("%s%s",
+                                                 parse_remote_path.stem(),
+                                                 kProfilingMetricsCsvSuffix);
+    std::string csv_remote_file_path = parse_remote_path.replace_extension(".csv").string();
+    auto        status = m_device->RetrieveFile(csv_remote_file_path,
+                                         settings.local_download_dir,
+                                         true,
+                                         csv_local_name);
+    if (!status.ok())
     {
-        LOGW("WARNING: Could not re-enable the compositor preemption: %s\n",
-             std::string(ret.message()).c_str());
+        return absl::NotFoundError("Failed to download the trace file: " + csv_remote_file_path);
     }
+    LOGI("RunReplayProfilingBinary(): Trace file %s downloaded to %s\n",
+         csv_remote_file_path.c_str(),
+         settings.local_download_dir.c_str());
+
+    LOGD("RunReplayProfilingBinary(): CLEANUP\n");
+    std::string clean_cmd = absl::StrFormat("shell rm -rf -- %s", remote_profiling_dir);
+    RETURN_IF_ERROR(m_device->Adb().Run(clean_cmd));
 
-    LOGD("RunReplayApk(): completed\n");
     return absl::OkStatus();
 }
 
-absl::Status DeviceManager::RunProfilingOnReplay(const std::string              &capture_path,
-                                                 const std::vector<std::string> &metrics,
-                                                 const std::string              &local_download_dir,
-                                                 const std::string              &gfxr_replay_flags)
+absl::Status DeviceManager::RunReplayApk(const GfxrReplaySettings &settings) const
 {
-    if (!m_device->IsAdrenoGpu())
+    LOGD("RunReplayApk(): Check settings before run\n");
+    auto ret_settings = ValidateGfxrReplaySettings(settings, m_device->IsAdrenoGpu());
```

Comment:
The [recommendation from Abseil](https://abseil.io/tips/181) is to treat absl::StatusOr as a pointer. So
- name it like you would a pointer. `validated_settings` would be my recommendation
- to access the value, use `*validated_settings` (after checking for .ok())

Additionally, I would prefer if auto was not used here so that the complexity of the return type is not hidden
```suggestion
    absl::StatusOr<GfxrReplaySettings> ret_settings = ValidateGfxrReplaySettings(settings, m_device->IsAdrenoGpu());
```
Abseil has some recommendations for when and when not to use `auto`: https://abseil.io/tips/232